src/bls/bls_aggregator.cpp

```diff
@@ -69,44 +93,62 @@ namespace {
         return result;
     }
 
-    std::vector<uint8_t> get_removal_msg_to_sign(
+    std::string dump_bls_exit_liquidation_response(const bls_exit_liquidation_response& item) {
+        std::string result =
+                "BLS exit response was:\n"
+                "\n"
+                "  - type:          {}\n"
+                "  - remove_pubkey: {}\n"
+                "  - timestamp:     {}\n"
+                "  - signature:     {}\n"
+                "  - msg_to_sign:   {}\n"_format(
+                        bls_exit_type_label(item.type),
+                        item.remove_pubkey,
+                        item.timestamp,
+                        item.signature,
+                        oxenc::to_hex(item.msg_to_sign.begin(), item.msg_to_sign.end()));
+        return result;
+    }
+
+    std::vector<uint8_t> get_exit_msg_to_sign(
             cryptonote::network_type nettype,
-            bls_aggregator::removal_type type,
+            bls_exit_type type,
             const bls_public_key& remove_pk,
             uint64_t unix_timestamp) {
-        // TODO(doyle): See BLSSigner::proofOfPossession
+        auto unix_timestamp_be = tools::encode_integer_be<32>(unix_timestamp);
         crypto::hash tag{};
-        std::vector<uint8_t> result;
-        if (type == bls_aggregator::removal_type::normal) {
-            tag = BLSSigner::buildTagHash(BLSSigner::removalTag, nettype);
-            result.reserve(tag.size() + remove_pk.size() + sizeof(unix_timestamp));
-            result.insert(result.end(), tag.begin(), tag.end());
-            result.insert(result.end(), remove_pk.begin(), remove_pk.end());
-            auto unix_timestamp_it = reinterpret_cast<uint8_t*>(&unix_timestamp);
-            result.insert(
-                    result.end(), unix_timestamp_it, unix_timestamp_it + sizeof(unix_timestamp));
+
+        if (type == bls_exit_type::normal) {
+            tag = BLSSigner::buildTagHash(BLSSigner::exitTag, nettype);
         } else {
-            assert(type == bls_aggregator::removal_type::liquidate);
+            assert(type == bls_exit_type::liquidate);
             tag = BLSSigner::buildTagHash(BLSSigner::liquidateTag, nettype);
-            result.reserve(tag.size() + remove_pk.size());
-            result.insert(result.end(), tag.begin(), tag.end());
-            result.insert(result.end(), remove_pk.begin(), remove_pk.end());
         }
+
+        // TODO(doyle): See BLSSigner::proofOfPossession
+        std::vector<uint8_t> result;
+        result.reserve(tag.size() + remove_pk.size() + unix_timestamp_be.size());
+        result.insert(result.end(), tag.begin(), tag.end());
+        result.insert(result.end(), remove_pk.begin(), remove_pk.end());
+        result.insert(
+                result.end(),
+                reinterpret_cast<uint8_t*>(unix_timestamp_be.begin()),
+                reinterpret_cast<uint8_t*>(unix_timestamp_be.end()));
```

Comment:
```suggestion
        auto *tsptr = reinterpret_cast<uint8_t*>(unix_timestamp_be.data());
        result.insert(result.end(), tsptr, tsptr + unix_timestamp_be.size());
```
(because I don't think std::array's begin/end iterators are strictly required to be pointers)

Reply:
Done